app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java

```diff
@@ -36,16 +40,48 @@
     private Context mContext;
     private BgGraphBuilder bgGraphBuilder;
     private BgReading mBgReading;
+    private static int lastTransactionId;
+    public PebbleSync(Context context){
+        this.mContext = context;
+        mBgReading = null;
+        init();
+    }
+    private void init() {
+        Log.i(TAG, "Initialising...");
+        Log.i(TAG, "configuring PebbleDataReceiver");
+
+        PebbleKit.registerReceivedDataHandler(mContext, new PebbleKit.PebbleDataReceiver(PEBBLEAPP_UUID) {
+            @Override
+            public void receiveData(final Context context, final int transactionId, final PebbleDictionary data) {
```

Comment:
so my question is, will this `receiveData` in between us getting new data on our end? or should this only get triggered once per new incoming BG.

Also, what exactly triggers this?